### PR TITLE
ACS-7350: add logging for HttpFileUploader

### PR DESCRIPTION
**Pull request by azakrzewski-hy**
[ACS-7350](https://hyland.atlassian.net/browse/ACS-7350)

Screenshot 2024-04-09 at 09 05 48

[ACS-7350]: https://hyland.atlassian.net/browse/ACS-7350?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mpichura — CHANGES_REQUESTED** [review]
```
Error:  Failures: 
Error:    HttpHxInsightStorageClientTest.testUploadDataFromInputStream:86 
Argument(s) are different! Wanted:
fileUploaderMock.upload(
    FileUploadRequest[file=File[data=Mock for InputStream, hashCode: 424974506], contentType=plain/text, storageLocation=Mock for URL, hashCode: 837267422]
);
-> at org.alfresco.hxi_connector.live_ingester.adapters.messaging.hx_insight.storage.HttpHxInsightStorageClientTest.testUploadDataFromInputStream(HttpHxInsightStorageClientTest.java:86)
Actual invocations have different arguments:
fileUploaderMock.upload(
    FileUploadRequest[file=File[data=Mock for InputStream, hashCode: 424974506], contentType=plain/text, storageLocation=Mock for URL, hashCode: 837267422],
    "node-ref"
);
-> at org.alfresco.hxi_connector.live_ingester.adapters.messaging.hx_insight.storage.HttpHxInsightStorageClient.upload(HttpHxInsightStorageClient.java:64)
```

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/FileUploader.java [hunk 1]
```diff
@@ -27,5 +27,8 @@
 
 public interface FileUploader
 {
+    @SuppressWarnings({"PMD.CloseResource", "PMD.PreserveStackTrace"})
```
Comment (tpage-alfresco) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/FileUploader.java:30]: I don't think we need this.

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/HttpFileUploader.java [hunk 2]
```diff
@@ -98,6 +98,17 @@ public void configure()
...
+        upload(fileUploadRequest, null);
+    }
+
+    @Retryable(retryFor = EndpointServerErrorException.class,
+            maxAttemptsExpression = "#{@integrationProperties.hylandExperience.storage.upload.retry.attempts}",
+            backoff = @Backoff(delayExpression = "#{@integrationProperties.hylandExperience.storage.upload.retry.initialDelay}",
+                    multiplierExpression = "#{@integrationProperties.hylandExperience.storage.upload.retry.delayMultiplier}"))
+    @Override
+    @SuppressWarnings({"PMD.CloseResource", "PMD.PreserveStackTrace"})
+    public void upload(FileUploadRequest fileUploadRequest, String nodeId)
```
Comment (tpage-alfresco) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/HttpFileUploader.java:111]: I think we should _only_ have this method, and get rid of the single parameter method. If this means updating a load of unit tests then that's fine - the tests should ideally call the same methods that the production code calls anyway.

**Comment by azakrzewski-hy** [comment]
> ```
> Error:  Failures: 
> Error:    HttpHxInsightStorageClientTest.testUploadDataFromInputStream:86 
> Argument(s) are different! Wanted:
> fileUploaderMock.upload(
>     FileUploadRequest[file=File[data=Mock for InputStream, hashCode: 424974506], contentType=plain/text, storageLocation=Mock for URL, hashCode: 837267422]
> );
> -> at org.alfresco.hxi_connector.live_ingester.adapters.messaging.hx_insight.storage.HttpHxInsightStorageClientTest.testUploadDataFromInputStream(HttpHxInsightStorageClientTest.java:86)
> Actual invocations have different arguments:
> fileUploaderMock.upload(
>     FileUploadRequest[file=File[data=Mock for InputStream, hashCode: 424974506], contentType=plain/text, storageLocation=Mock for URL, hashCode: 837267422],
>     "node-ref"
> );
> -> at org.alfresco.hxi_connector.live_ingester.adapters.messaging.hx_insight.storage.HttpHxInsightStorageClient.upload(HttpHxInsightStorageClient.java:64)
> ```

done in [260c2e1](https://github.com/Alfresco/hxinsight-connector/pull/274/commits/260c2e120a8371465126244b8ca2362e94ece209)

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/FileUploader.java (hunk 1 above)
Comment (azakrzewski-hy) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/FileUploader.java]: done in [260c2e1](https://github.com/Alfresco/hxinsight-connector/pull/274/commits/260c2e120a8371465126244b8ca2362e94ece209)

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/HttpFileUploader.java (hunk 2 above)
Comment (azakrzewski-hy) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/HttpFileUploader.java:100]: done in [260c2e1](https://github.com/Alfresco/hxinsight-connector/pull/274/commits/260c2e120a8371465126244b8ca2362e94ece209)

**Review by tpage-alfresco — APPROVED** [review]
Looks good. Thanks Adam!

**Review by mpichura — APPROVED** [review]
LGTM

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/HttpFileUploader.java [hunk 3]
```diff
@@ -111,6 +111,7 @@ STORAGE_LOCATION_HEADER, wrapRawToken(fileUploadRequest.storageLocation()),
                     .withHeaders(headers)
                     .withBody(fileData)
                     .request();
+            log.atDebug().log("Upload :: PDF rendition of the node: {} to presignedUrl: {}", nodeId, fileUploadRequest.storageLocation().getPath());
```
Comment (krdabrowski) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/HttpFileUploader.java:114]:
```suggestion
            log.atDebug().log("Upload :: PDF rendition of the node: {} successfully uploaded to pre-signed URL: {}", nodeId, fileUploadRequest.storageLocation().getPath());
```

**Review by krdabrowski — APPROVED** [review]
Overall LGTM, I only propose one change to log message as it's a bit ambiguous IMO

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/HttpFileUploader.java (hunk 3 above)
Comment (azakrzewski-hy) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/HttpFileUploader.java]: log message updated

**Pull request merged**